cloudpickle/cloudpickle.py

```diff
@@ -141,22 +142,45 @@ def _whichmodule(obj, name):
     return None
 
 
-def _is_global(obj, name=None):
+if sys.version_info[:2] < (3, 7):
+    # Workaround bug in old Python versions: prior to Python 3.7, T.__module__
+    # would always be set to "typing" even when the TypeVar T would be defined
+    # in a different module.
+    #
+    # For such older Python versions, we treat all TypeVar instances as
+    # non-importable TypeVar instances to avoid this issue.
```

Comment:
This is not exactly true: for older `Python` versions, because `obj.__module__` is set to a wrong value, we patch `getattr(obj, "__module__")` to return `None` in order to trigger an extensive search for `obj` inside all imported modules during `_whichmodule` execution.

These `TypeVar`s do end up getting pickled as importable objects.

Reply:
Thanks let me push a fix.